WalletSdk/src/main/java/com/spruceid/wallet/sdk/StorageManager.kt

```diff
@@ -0,0 +1,149 @@
+import android.content.Context
+import android.util.Base64
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.*
+import androidx.datastore.preferences.preferencesDataStoreFile
+import com.spruceid.wallet.sdk.KeyManager
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.map
+
+private class DataStoreSingleton private constructor(context: Context) {
+    val dataStore: DataStore<Preferences> = store(context, "default")
+
+    companion object {
+        private const val FILENAME_PREFIX = "datastore_"
```

Comment:
For the prefix and the encryption key, could we use something sprucekit specific to avoid name clashes when someone imports this library into their application? Maybe:

* `sprucekit/datastore/` directories to store the files in?
* `sprucekit-datastore` for the encryption key?

Reply:
@todd-spruceid WDYT doing something similar in mobile-sdk-swift for consistency?

Reply:
I'd be inclined to suggest just `sprucekit` for the directory, but regardless, it probably makes sense to do the same in `-swift`, mostly for ease of maintenance.